### PR TITLE
doc: layotto rpc code analysis

### LINKED ISSUE #424: 源码解析: 处理 rpc 请求
Note: since this is a Chinese community activity,I wrote it in Chinese.
 
### 介绍
阅读源码，分析 Layotto 处理rpc请求的流程。

Layotto “寄生”在 MOSN 里，rpc请求到达sidecar时，其实是先经过 MOSN、MOSN通过[grpcFilter](https://github.com/mosn/mosn/blob/master/pkg/filter/network/grpc/filter.go) 把流量转给 Layotto ，Layotto 解包、处理之后，再通过一对虚拟连接(net.Pipe) 将流量转给MOSN, MOSN做流量转发

### 参考解析

- 处理 rpc 请求的流程是什么样的

###  参考阅读
[进行RPC调用--helloworld](https://mosn.io/layotto/#/zh/start/rpc/helloworld)
[进行RPC调用--Dubbo JSON RPC Example](https://mosn.io/layotto/#/zh/start/rpc/dubbo_json_rpc)
https://mosn.io/layotto/#/zh/building_blocks/rpc/reference
https://mosn.io/layotto/#/zh/design/rpc/rpc%E8%AE%BE%E8%AE%A1%E6%96%87%E6%A1%A3

### 源码解析文章的格式
可以参考之前 “MOSN 源码解析活动”的文章格式 https://mosn.io/blog/code/

### PR DESCRIPTION
**Pull request by rayowang**
**What this PR does**:
layotto rpc code analysis 

**Which issue(s) this PR fixes**:
Fixes #424 

**Special notes for your reviewer**:
Because it took some time to repair dubbo-go-sample before, and now the deadline is up, so I submit a version first. I hope that the tutor and maintainers can help to see if there is anything that needs to be supplemented and improved. In addition, I will add a summary and further improvement over the weekend.

Further more, I would like to make a suggestion that the source code analysis document should be similar to other open source projects. All source code analysis can create a separate code directory under the blog directory, and then create another subdirectory under the code directory to submit markdown files and pictures and other related files. 

**Does this PR introduce a user-facing change?**:
NONE

### CODE REVIEW

**Comment by rayowang** [comment]
> Please add this doc to the sidebar. See https://mosn.io/layotto/#/zh/development/contributing-doc?id=%e6%96%b0%e5%a2%9e%e6%96%87%e6%a1%a3%e9%9c%80%e8%a6%81%e5%81%9a%e5%93%aa%e4%ba%9b%e4%ba%8b image

The sidebar has been updated, and by the way, all the current source code analysis documents have been moved to /zh/blog/code.

**Review by zhenjunMa — DISMISSED** [review]
LGTM

File: docs/zh/blog/code/layotto-rpc/index.md [hunk 1]
```diff
@@ -0,0 +1,921 @@
+---
+title: "Layotto 源码解析 —— 处理 RPC 请求"
+linkTitle: "Layotto 源码解析 —— 处理 RPC 请求"
+date: 2022-04-21
+aliases: "zh/blog/code/layotto-rpc"
+author: "[王志龙](https://github.com/rayowang)"
+description: >
+  本文主要以 Dubbo Json RPC 为例来分析 Layotto RPC 处理流程
+---
```
Comment (seeflood) [inline docs/zh/blog/code/layotto-rpc/index.md:9]: This header won't be rendered correctly:
image
Note: layotto use docsify to build the doc site, see https://mosn.io/layotto/#/zh/development/contributing-doc?id=%e6%96%87%e6%a1%a3%e7%ab%99%e7%82%b9%e8%af%b4%e6%98%8e

You can run this command to start a doc site locally:
```
docsify serve docs
```

To fix this issue, you can use the quote syntax, like https://mosn.io/layotto/#/zh/blog/flowcontrol_code_analyze
image

**Review by seeflood — CHANGES_REQUESTED** [review]

File: docs/zh/blog/code/layotto-rpc/index.md [hunk 2]
```diff
@@ -0,0 +1,921 @@
...
+        return 0, nil, err
+    }
+    ...
+}
+```
+最终收到返回数据：
+{"jsonrpc":"2.0","id":9527,"result":{"id":"113","name":"Moorse","age":30,"time":703394193,"sex":"MAN"}}
+
+## 总结
+Layotto RPC 处理流程涉及 GRPC、Dapr、Mosn 等相关的知识，整体流程较长，不过单纯看 Layotto 针对 Mosn 抽象的轻量 RPC 框架还是比较清晰和简单的，与 Mosn 集成的方式也比较新颖，值得进一步研读。至此 Layotto RPC 请求处理就分析完了，时间有限，没有进行一些更全面和深入的剖析，如有纰漏之处，欢迎指正，联系方式：rayo.wangzl@gmail.com。另外在此也希望大家能踊跃参与源码分析和开源社区来，一起学习，共同进步。
```
Comment (seeflood) [inline docs/zh/blog/code/layotto-rpc/index.md:919]: @MoonShining Haha, your code is highly appreciated :)
Could u help review this PR ?

File: docs/zh/blog/code/layotto-rpc/index.md (hunk 1 above)
Comment (rayowang) [reply docs/zh/blog/code/layotto-rpc/index.md]: Oh, I thought it was the same blog generator used by mosn, so I wrote it like this. I just fixed it.

**Review by zhenjunMa — APPROVED** [review]

**Review by seeflood — APPROVED** [review]
lgtm. Thanks for your detailed article !

**Pull request merged**